src/PackageParser.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
+ *
+ * This source code is licensed under the Apache 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { readJsonSync } from 'fs-extra';
+
+/**
+ * The PackageParser class.
+ * It provides methods to extract information form a package.json file.
+ */
+class PackageParser {
+  /**
+   * Constructor of the PackageParser
+   */
+  constructor() {
+    this.content = null;
+    this.linkableDependencies = [];
+  }
+
+  /**
+   * Initializes the package parser by parsing the package.json at the specified path.
+   * @param {string} path The path of the package.
+   * @return {PackageParser}
+   */
+  parse(path) {
+    try {
+      this.content = readJsonSync(`${path}/package.json`);
+    } catch (e) {
+      this.content = null;
+      throw new Error(`Invalid package at ${path}`);
```

Comment:
Why do you throw an error here if you catch one before? I would only log it instead. Or does the process need to be quit here?

Reply:
I wanted to have a custom error message and that describes what happened. It's handling is implemented at the code where the `PackageParser` is used.